src/Claims.ts

```diff
@@ -0,0 +1,206 @@
+import { Identity } from '~/api/entities';
+import { modifyClaims, ModifyClaimsParams } from '~/api/procedures';
+import { TransactionQueue } from '~/base';
+import { Context } from '~/context';
+import { didsWithClaims, scopesByIdentity } from '~/middleware/queries';
+import { ClaimTypeEnum, Query } from '~/middleware/types';
+import { ClaimData, ClaimScope, ClaimType, Ensured, IdentityWithClaims, ResultSet } from '~/types';
+import { ClaimOperation } from '~/types/internal';
+import { calculateNextKey, createClaim, removePadding, valueToDid } from '~/utils';
+
+/**
+ * Handles all Claims related functionality
+ */
+export class Claims {
+  private context: Context;
+
+  /**
+   * @hidden
+   */
+  constructor(context: Context) {
+    this.context = context;
+  }
+
+  /**
+   * Add claims to identities
+   *
+   * @param args.claims - array of claims to be added
+   */
+  public addClaims(args: Omit<ModifyClaimsParams, 'operation'>): Promise<TransactionQueue<void>> {
+    return modifyClaims.prepare({ ...args, operation: ClaimOperation.Add }, this.context);
+  }
+
+  /**
+   * Edit claims associated to identities (only the expiry date can be modified)
+   *
+   * * @param args.claims - array of claims to be edited
+   */
+  public editClaims(args: Omit<ModifyClaimsParams, 'operation'>): Promise<TransactionQueue<void>> {
+    return modifyClaims.prepare({ ...args, operation: ClaimOperation.Edit }, this.context);
+  }
+
+  /**
+   * Revoke claims from identities
+   *
+   * @param args.claims - array of claims to be revoked
+   */
+  public revokeClaims(
+    args: Omit<ModifyClaimsParams, 'operation'>
+  ): Promise<TransactionQueue<void>> {
+    return modifyClaims.prepare({ ...args, operation: ClaimOperation.Revoke }, this.context);
+  }
+
+  /**
+   * Retrieve all claims issued by the current identity
```

Comment:
This function should accept an identity as a parameter, and default to the current one

`Retrieve all claims issued by an Identity`

Reply:
Also, add `@note uses the middleware` to the comments